custom comparator support in calculate_numeric_gap

Gasoonjia opened the pull request:
Summary: Enable users to define their own metrics for numeric gap.

Differential Revision: D87840446

winskuo-quic commented on devtools/inspector/_inspector.py:
```diff
@@ -1416,7 +1419,10 @@ def calculate_numeric_gap(
         Args:
-            distance: The metrics the inspector will use for gap calculation. Should be one of "MSE", "L1" and "SNR".
+            distance: The metrics the inspector will use for gap calculation. Can be either:
+                - A string: one of "MSE", "L1", or "SNR" for built-in comparators.
```
Should this be `- A string: one of "MSE", "L1", "SNR", or user-defined comparators`?

winskuo-quic reviewed (commented):
Thanks for supporting this feature and providing users more flexibility!

Gasoonjia committed:
custom comparator support in calculate_numeric_gap (#15969)

Summary:

Enable users to define their own metrics for numeric gap.

Differential Revision: D87840446
